### PR TITLE
[SGP-8096] Update django-currency-rates to support Python 3

### PR DESCRIPTION
**Pull request by caffodian**
- Forked Joel's fork
- cleaned up Py 2/3 stuff (urlopen via six, items vs. iteritems, unicode models)
- added Django 2.x stuff (on_delete required for ForeignKey, which is a no-op since that's the default in 1.x anyway) 
- fixed test Django project to have modern settings
- added initial migration for post-south world 
- run tests for future django/python versions using tox
- run tox on circleci 

I'd really like to publish this on pypi as django-currency-rates-redux or something, but that's  a future convo.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+version: 2
+jobs:
+  build:
+    docker:
+        - image: kaikuehne/pyenv:alpine3.10
```
Comment (caffodian) [inline .circleci/config.yml:5]: We should probably have our own version of this image in the startupgrind org.  (eventually)

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+                pyenv local  2.7.16 3.7.4
+                pip install --upgrade pip
+                pip install tox tox-pyenv
+            shell: /bin/bash
+        - save_cache:
+              key: v1-project-{{ arch }}-{{ checksum "tox.ini" }}
+              paths:
+                  - /root/pyenv
+        - run:
+            command: tox
```
Comment (caffodian) [inline .circleci/config.yml:34]: This workflow needs a lot of work, but it works for now.  It's copied from my bad attempt at getting percy working with circleci.

File: currency_rates/management/commands/load_rates.py [hunk 3]
```diff
@@ -35,7 +35,7 @@ def handle_noargs(self, **options):
 
         ExchangeRate.objects.filter(date=date).delete()
 
-        for code, rate in data['rates'].iteritems():
+        for code, rate in data['rates'].items():
```
Comment (caffodian) [inline currency_rates/management/commands/load_rates.py:38]: the iterator distinction is irrelevant in this case.

File: currency_rates/tests.py [hunk 4]
```diff
@@ -170,10 +171,10 @@ def test_load_rates_ignore_non_existent(self):
         # create currency
         Currency.objects.create(code="USD")
 
-        with mock.patch('urllib2.urlopen') as mock_urlopen:
+        with mock.patch('currency_rates.management.commands.load_rates.urlopen') as mock_urlopen:
```
Comment (caffodian) [inline currency_rates/tests.py:174]: dunno, mocking six doesn't work here (probably happened too late?) so i gave up and did this instead

File: .circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+                - v1-project-
+        - run:
+            command: |
+                pyenv --version
+                pyenv install --list
+                pyenv install --keep --skip-existing 2.7.16
+                pyenv install --keep --skip-existing 3.7.4
+                pyenv rehash
+                pyenv versions
+                pyenv local  2.7.16 3.7.4
```
Comment (rbevers) [inline .circleci/config.yml:25]: What does this do? Can't it only handle a single version at a time?

File: currency_rates/management/commands/load_rates.py [hunk 6]
```diff
@@ -21,7 +21,7 @@ def handle_noargs(self, **options):
         base_currency = getattr(settings, "CURRENCY_RATES_DEFAULT_CODE", 'EUR')
 
         url = CURRENT_RATES_URL + "?app_id=" + app_id
-        f = urllib2.urlopen(url)
+        f = urlopen(url)
```
Comment (rbevers) [inline currency_rates/management/commands/load_rates.py:24]: 👍

File: example/example/settings.py [hunk 7]
```diff
@@ -157,4 +157,4 @@
 }
 
 import os
-OPENEXCHANGERATES_APP_ID = os.environ['OPENEXCHANGERATES_APP_ID']
+OPENEXCHANGERATES_APP_ID = os.environ.get('OPENEXCHANGERATES_APP_ID','')
```
Comment (rbevers) [inline example/example/settings.py:160]: 👍

**Review by rbevers — COMMENTED** [review]
Looks good, except for the `pyenv` command I don't understand. Happy to approve after I know.

File: .circleci/config.yml (hunk 5 above)
Comment (caffodian) [reply .circleci/config.yml:25]: https://github.com/pyenv/pyenv/blob/master/COMMANDS.md#pyenv-local-advanced

it sets up `python` => 2.7
`python2.7` 
`python3.7`


This isn't really needed for pyenv-tox to work, but is convenient if you ever do need to ssh into the circleci job.

Comment (rbevers) [reply .circleci/config.yml:25]: Cool. TIL... was going off this...

```bash
$ pyenv local --help
Usage: pyenv local <version>
       pyenv local --unset

Sets the local application-specific Python version by writing the
version name to a file named `.python-version'.

When you run a Python command, pyenv will look for a `.python-version'
file in the current directory and each parent directory. If no such
file is found in the tree, pyenv will use the global Python version
specified with `pyenv global'. A version specified with the
`PYENV_VERSION' environment variable takes precedence over local
and global versions.

<version> should be a string matching a Python version known to pyenv.
The special version string `system' will use your default system Python.
Run `pyenv versions' for a list of available Python versions.
```

**Review by rbevers — APPROVED** [review]
👍 :shipit: Awake to the modern era of finances.

**Pull request merged**